test/mpi/f77/ext/c2f2c.c

```diff
@@ -172,7 +172,9 @@ MPI_Fint c2frequest_(MPI_Fint * request)
     MPI_Request req = MPI_Request_f2c(*request);
     MPI_Status status;
     int flag;
-    MPI_Test(&req, &flag, &status);
+    do {
+        MPI_Test(&req, &flag, &status);
+    } while (!flag);
```

Comment:
Will this be equivalent to `MPI_Wait`?

Reply:
Hmm. Maybe.

Reply:
Do you want me to replace this with `MPI_Wait`?

Reply:
At least try it and test. If it is the same, then `MPI_Wait` is more explicitly better.

Reply:
By looking back the code again, the right fix might be something like
```C
MPI_Test(&req, &flag, &status);
if (flag) {
    MPI_Test_canceled(...);
}
```
because if I look at the later part of the code, it's returning the request object to the caller, and the the caller calls `mpi_wait`.
```
      call mpi_irecv( 0, 0, MPI_INTEGER, MPI_ANY_SOURCE, MPI_ANY_TAG,
     $     MPI_COMM_WORLD, req, ierr )
      call mpi_cancel( req, ierr )
      errs = errs + c2frequest( req )
      call mpi_wait( req, MPI_STATUS_IGNORE, ierr )
```

Reply:
Done.